wrappers/hnygrpc/grpc.go

```diff
@@ -72,16 +74,26 @@ func addFields(ctx context.Context, info *grpc.UnaryServerInfo, handler grpc.Una
 	span.AddField("handler.name", handlerName)
 	span.AddField("handler.method", info.FullMethod)
 
-	md, ok := metadata.FromIncomingContext(ctx)
+	pr, ok := peer.FromContext(ctx)
 	if ok {
-		if val, ok := md["content-type"]; ok {
-			span.AddField("request.content_type", val[0])
-		}
-		if val, ok := md[":authority"]; ok {
-			span.AddField("request.header.authority", val[0])
+		if pr.Addr != net.Addr(nil) {
+			span.AddField("request.remote_addr", pr.Addr.String())
 		}
-		if val, ok := md["user-agent"]; ok {
-			span.AddField("request.header.user_agent", val[0])
+	}
+
+	headersToFields := map[string]string{
```

Comment:
This should be made a const so it doesn't get redeclared for each call to `addFields`.

Reply:
I would love to, but Go doesn't support complex types like maps and slices to be declared directly as constants. 

Options that if you squint kind of look like a constant:

* a function -  We could have a private function that returns a map, but my understanding is that this would still be relentlessly allocating new memory for the local variable.

```go
func getHeadersToFields() map[string]string {
	return map[string]string{
		"content-type":      "request.content_type",
		":authority":        "request.header.authority",
		"user-agent":        "request.header.user_agent",
		"x-forwarded-for":   "request.header.x_forwarded_for",
		"x-forwarded-proto": "request.header.x_forwarded_proto",
	}
}
```

* a package var - We could declare it as a variable in this package and be diligent about not mutating it.

```go
package hnygrpc

import (
	… 
)

var headersToFields = map[string]string{
	"content-type":      "request.content_type",
	":authority":        "request.header.authority",
	"user-agent":        "request.header.user_agent",
	"x-forwarded-for":   "request.header.x_forwarded_for",
	"x-forwarded-proto": "request.header.x_forwarded_proto",
}
```

Reply:
Ah, I'm sure I've suggested making a map a const before too 😄 

I like the var option, with comment to say do not update 👍🏻

Reply:
private package var is a fairly common solution to this, yes. 

And just to be pedantic about it, Go doesn't allocate memory on the heap for a locally-defined map that isn't modified, but it does reserve space for it on the stack and has to initialize it. So it costs CPU cycles but doesn't add memory pressure or allocations.

Reply:
> … just to be pedantic …

I appreciate the details!

![The More You Know](https://user-images.githubusercontent.com/517302/186471051-b8ad893f-bb10-41f9-85de-6bad532692f5.gif)

Reply:
Local variable moved up to a package variable ([194b95f](https://github.com/honeycombio/beeline-go/pull/341/commits/194b95f72b8817366852076fcc536e9278df5be0)).